lib/Dialect/AIE/Transforms/AIECoreToStandard.cpp

```diff
@@ -65,7 +65,8 @@ struct AIEDebugOpToStdLowering : public OpConversionPattern<DebugOp> {
 
     std::string funcName = "debug_i32";
     auto func = module.lookupSymbol<func::FuncOp>(funcName);
-    assert(func && "Could not find the intrinsic function!");
+    if (!func)
+      return module.emitOpError("Could not find the intrinsic function!");
```

Comment:
It's not clear to me that this is a user-visible OpError?  I assume your code is creating the debug_i32 intrinsic somewhere else?  In that case, an assert is actually more appropriate, since there's nothing about the user's input that they could change to fix it.

Reply:
I am not actually very familiar with this transform so I don't know where the intrinsic function is created. My reasoning for removing the assert() within the pass was to use MLIR's conversion pattern failure system for more graceful failing and also ensure that it would fail even without asserts() being enabled.

Reply:
Got it... Basically, if this is an internal consistency check, then assert() is appropriate.  If its an assumption about valid inputs, then opError is appropriate.  Unfortunately, between passes, this can become tricky, since pass A may generate something that pass B needs:  So is that an 'internal consistency check', because we always run pass A before pass B, or is an assumption?  Both types of messaging can be confusing, but I'd probably err on the side of emitOpError.

Reply:
I see! I will check whether there is another pass that always runs before this one, in which case I will switch back to using assert(). But if the pass can be used as a standalone then I will leave emitOpError. Thank you for the insights!